pkg/network/tracer/connection/tracer.go

```diff
@@ -304,6 +308,12 @@ func (t *tracer) GetConnections(buffer *network.ConnectionBuffer, filter func(*n
 	// Iterate through all key-value pairs in map
 	key, stats := &netebpf.ConnTuple{}, &netebpf.ConnStats{}
 	seen := make(map[netebpf.ConnTuple]struct{})
+	// connsByTuple is used to detect whether we are iterating over
+	// a connection we have previously seen. This can happen when
+	// ebpf maps are being iterated over and deleted at the same time.
```

Comment:
Is this concurrency within our power to control? If so, would we be able to solve it with a mutex around iterate / delete?

Reply:
Unfortunately no, we have had a long discussion on slack about this, and there are better solutions being discussed. This is just to make as many fixes as we can right now.